js_modules/dagit/packages/core/src/assets/AutoMaterializePolicyPage/getEvaluationsWithEmptyAdded.tsx

```diff
@@ -15,34 +15,46 @@ export const getEvaluationsWithEmptyAdded = ({
   isFirstPage,
   isLastPage,
 }: Config) => {
-  if (isLoading || !currentEvaluationId) {
+  if (isLoading) {
     return [];
   }
+
   const evalsWithSkips = [];
-  let current = isFirstPage ? currentEvaluationId : evaluations[0]?.evaluationId || 1;
-  evaluations.forEach((evaluation, i) => {
-    const prevEvaluation = evaluations[i - 1];
-    if (evaluation.evaluationId !== current) {
-      evalsWithSkips.push({
-        __typename: 'no_conditions_met' as const,
-        evaluationId: current,
-        amount: current - evaluation.evaluationId,
-        endTimestamp: prevEvaluation?.timestamp ? prevEvaluation?.timestamp - 60 : ('now' as const),
-        startTimestamp: evaluation.timestamp + 60,
-      });
-    }
-    evalsWithSkips.push(evaluation);
-    current = evaluation.evaluationId - 1;
-  });
-  if (isLastPage && current > 0) {
+
+  if (evaluations.length) {
+    let current =
+      isFirstPage && currentEvaluationId !== null
+        ? currentEvaluationId
+        : evaluations[0]?.evaluationId || 1;
+
+    evaluations.forEach((evaluation, i) => {
+      const prevEvaluation = evaluations[i - 1];
+      if (evaluation.evaluationId !== current) {
+        evalsWithSkips.push({
+          __typename: 'no_conditions_met' as const,
+          evaluationId: current,
+          amount: current - evaluation.evaluationId,
+          endTimestamp: prevEvaluation?.timestamp
+            ? prevEvaluation?.timestamp - 60
+            : ('now' as const),
+          startTimestamp: evaluation.timestamp + 60,
+        });
+      }
+      evalsWithSkips.push(evaluation);
+      current = evaluation.evaluationId - 1;
+    });
+  }
+
+  if (isLastPage) {
     const lastEvaluation = evaluations[evaluations.length - 1];
     evalsWithSkips.push({
       __typename: 'no_conditions_met' as const,
-      evaluationId: current,
-      amount: current - 0,
+      evaluationId: 0,
+      amount: 0,
```

Comment:
Do we not want to show the amount anymore?

Reply:
Isn't this item the "Before [...]" case where there are none prior to that?

Reply:
The backend doesn't return skips, this function is trying to detect how many skips there are. It's possible there are none before because they're all skips.